core/trino-parser/src/test/java/io/trino/sql/parser/TestSqlParser.java

```diff
@@ -2523,36 +2522,41 @@ public void testShowStatsForQuery()
 
         for (String fullName : tableNames) {
             QualifiedName qualifiedName = makeQualifiedName(fullName);
+
+            // Simple SELECT
             assertStatement(format("SHOW STATS FOR (SELECT * FROM %s)", qualifiedName),
-                    createShowStats(qualifiedName, ImmutableList.of(new AllColumns()), Optional.empty()));
+                    new ShowStats(
+                            new TableSubquery(simpleQuery(new Select(false, ImmutableList.of(new AllColumns())),
+                                    new Table(qualifiedName),
+                                    Optional.empty(),
+                                    Optional.empty()))));
+
+            // SELECT with predicate
             assertStatement(format("SHOW STATS FOR (SELECT * FROM %s WHERE field > 0)", qualifiedName),
-                    createShowStats(qualifiedName,
```

Comment:
Note that this is all being changed in this PR: https://github.com/trinodb/trino/pull/7861

Reply:
dropped